### DISCUSSION TITLE
Define transition by function

### DISCUSSION THREAD

**Discussion by alnaum** [nuxt/nuxt#16900] [Questions]
Hello!

Does Nuxt 3 have the feature to define page transition name by function like in Nuxt 2?

```
export default {
  transition(to, from) {
    if (!from) {
      return 'slide-left'
    }
    return +to.query.page < +from.query.page ? 'slide-right' : 'slide-left'
  }
}
```
If not, how can I change the name of the transition by some condition? 
I tried rewrite route meta before manual transition:

```
 const router = useRouter();
 router.currentRoute.value.meta.pageTransition = {
    ...router.currentRoute.value.meta.pageTransition,
    name: newIndex > index ? "slide-left" : "slide-right",
 };
 router.push(allRoutes[newIndex]);
```
This seems to work on the local dev server, but not on the deployed app.

**Comment by acidjazz** [comment]
Related https://github.com/nuxt/framework/issues/7428
